### PR TITLE
Handle additional container cases

### PR DESCRIPTION
**Pull request by drewkerrigan**
This addresses https://dcosjira.atlassian.net/projects/DCOS/issues/DCOS-301

### CODE REVIEW

**Comment by drewkerrigan** [comment]
@sargun please review when possible

File: include/mesos_state.hrl [hunk 1]
```diff
@@ -20,7 +20,7 @@
 -type role() :: binary().
 -type slave_id() :: binary().
 -type hostname() :: binary().
--type protocol() :: tcp | udp.
+-type protocol() :: tcp | udp | client.
```
Comment (sargun) [inline include/mesos_state.hrl]: Can you ping the folks and ask them why they have client as a new type?

File: src/mesos_state_client.erl [hunk 2]
```diff
@@ -301,36 +306,15 @@ port_mapping(#{container_port := ContainerPort, host_port := HostPort, protocol
...
-docker(_Docker = #{force_pull_image := ForcePullImage, image := Image, network := <<"HOST">>}) ->
-  #docker{force_pull_image = ForcePullImage, image = Image, network = host, port_mappings = []}.
+force_pull_image(#{force_pull_image := ForcePullImage}) -> ForcePullImage;
+force_pull_image(_) -> false.
+
+docker(#{image := Image, network := Network} = Docker) ->
+    #docker{
+       force_pull_image = force_pull_image(Docker),
+       image = Image,
+       network = list_to_atom(string:to_lower(binary_to_list(Network))),
```
Comment (sargun) [inline src/mesos_state_client.erl]: Do we want to do list_to_existing_atom to prevent memory leak?

File: include/mesos_state.hrl (hunk 1 above)
Comment (drewkerrigan) [inline include/mesos_state.hrl]: I don't think the user intentionally created the new type, tracing where the "client" came from will take some digging. We have the marathon json to replicate the issue so I think I should probably start there.

File: src/mesos_state_client.erl (hunk 2 above)
Comment (drewkerrigan) [inline src/mesos_state_client.erl]: Probably a good call, yes.

**Comment by sargun** [comment]
Lgtm

**Comment by sargun** [comment]
Plz squash the deps.

**Comment by drewkerrigan** [comment]
Done

**Pull request merged**